src/Akeneo/Tool/Component/BatchQueue/Queue/PublishJobToQueue.php

```diff
@@ -160,23 +115,22 @@ private function createJobParameters(JobInterface $job, JobInstance $jobInstance
         return $this->jobParametersFactory->create($job, $rawParameters);
     }
 
-    private function validateJobParameters(JobInterface $job, JobInstance $jobInstance, JobParameters $jobParameters, string $code) : void
+    private function validateJob(JobInterface $job, JobInstance $jobInstance, JobParameters $jobParameters, string $code): void
     {
         // We merge the JobInstance from the JobManager EntityManager to the DefaultEntityManager
         // in order to be able to have a working UniqueEntity validation
         $this->entityManager->merge($jobInstance);
-        $errors = $this->jobParametersValidator->validate($job, $jobParameters, ['Default', 'Execution']);
-
-        if (count($errors) > 0) {
-            throw new \RuntimeException(
-                sprintf(
-                    'Job instance "%s" running the job "%s" with parameters "%s" is invalid because of "%s"',
-                    $code,
-                    $job->getName(),
-                    print_r($jobParameters->all(), true),
-                    $this->getErrorMessages($errors)
-                )
-            );
+
+        $jobInstanceViolations = $this->validator->validate($jobInstance, new JobInstanceConstraint());
```

Comment:
You didn't need to give the JobInstanceConstraint, it's better to validate the whole job
```suggestion
        $jobInstanceViolations = $this->validator->validate($jobInstance);
```